src/cargo/core/registry.rs

```diff
@@ -454,7 +454,7 @@ impl<'cfg> Registry for PackageRegistry<'cfg> {
                     // If we don't have an override then we just ship
                     // everything upstairs after locking the summary
                     (None, Some(source)) => {
-                        for patch in patches.iter() {
+                        for patch in &patches {
```

Comment:
Historically, I believe @alexcrichton preferred the `.iter` form, because it is more explicit. I personally also favor `.iter()`. 

However this definitely contradicts clippy's default, and there's certain benefit in sticking with defaults...

@alexcrichton do you have any plan here? Should we

0) Persuade clippy folks that `.iter()` is better and switch clippy's default :) 
1) Give up on our preferences and just switch from `.iter` to `&` everywhere? 
2) Configure `clippy` in Cargo.toml, so that it does not give this warning
3) Continue to reject PRs with this change once in a while

Reply:
> Continue to reject PRs with this change once in a while
I am sorry. I should have checked whether this has come up before.

Reply:
Once this is decided I am happy to redo the other changes in this PR and re submit.

Reply:
Yeah I personally prefer `.iter()` as I find `for x in &foo` to be less clear and also slightly less readable, but again that's mostly personal preference

Reply:
@alexcrichton @Eh2406 let's try to add an exception for this rule to Cargo.toml than?

Although we don't run Clippy on Cargo on CI, it'll help people who do so locally!

Reply:
I will re do this pr without all of them, next time I have spare cycles. :-)

Reply:
Great, thanks @Eh2406!

Reply:
Yeah I'd be fine adding an exception!